### DISCUSSION TITLE
how to move branch with many images betwen mind maps files

### DISCUSSION THREAD

**Discussion by alexalxs** [freeplane/freeplane#1455] [General]
I have a situation as follows: I created the mind map m1.mm which has a branch with a total of 1000 images. Now I need to cut this branch to add it to the m2.mm mind map but the images are broken. What's the solution? Would I have to add the 1000 images manually so that the references to the files are correct or is there any way to do this automatically?

**Comment by macmarrum** [comment]
You probably used relative paths for images, therefore if you save your m2.mm in the same directory as m1.mm, your images should be visible.

If m2.mm is in a different directory, you'll need to change the relative paths. Freeplane has a nice Scripting API, which allows to automate a lot of operations, including this one.

The following pages might be of interest to you:
- https://docs.freeplane.org/scripting/Scripting.html
- https://docs.freeplane.org/scripting/api-groovy-tutorial.html
- https://docs.freeplane.org/api/org/freeplane/api/ExternalObject.html

**Reply by beccon4** [reply]
well in other programs I'd just copy and paste without taking care what is where. That should be possible here as well.,

**Reply by EdoFro** [reply]
maybe you're right. may be you can implement it.
